components/collector/src/source_collectors/owasp_zap/security_warnings.py

```diff
@@ -40,8 +40,8 @@ def __alert_instance_entity(self, ids, entity_kwargs, alert_instance) -> Entity:
         """Create an alert instance entity."""
         method = alert_instance.findtext("method", default="")
         uri = self.__stable_url(hashless(URL(alert_instance.findtext("uri", default=""))))
-        key = md5_hash(f"{':'.join(ids)}:{method}:{uri}")
-        old_key = md5_hash(f"{':'.join(ids[1:])}:{method}:{uri}")
+        key = md5_hash(f"{':'.join(ids[:-1])}:{method}:{uri}")
+        old_key = md5_hash(f"{':'.join(ids)}:{method}:{uri}")
```

Comment:
Why are you adding the alert to the old key?

Reply:
This allows Quality-time to keep track of the entity's status even though its key changed. See https://github.com/ICTU/quality-time/blob/d5f528f06680f06e898265be909073ff1bb4d280/components/server/src/model/source.py#L38